eradiate/radprops/mie.py

```diff
@@ -0,0 +1,318 @@
+"""
+Mie scattering properties computation.
+"""
+import typing as t
+
+import miepython
+import numpy as np
+import pint
+import xarray as xr
+
+
+from ..units import unit_registry as ureg
+
+_DEFAULT_MU = np.linspace(-1, 1, 201)
+
+
+@ureg.wraps(ret=None, args=("nm", None, None, "dimensionless"), strict=False)
+def compute_properties(
+    w: t.Union[ureg.Quantity, np.ndarray],
+    rdist: xr.DataArray,
+    m: xr.DataArray,
+    mu: t.Union[ureg.Quantity, np.ndarray] = _DEFAULT_MU,
+) -> xr.DataArray:
+    """
+    Compute multi-wavelength radiative properties of spherical particles with
+    a radius distribution.
+
+    Parameters
+    ----------
+    w: :class:`~pint.Quantity` or :class:`~numpy.ndarray`
+        Wavelength values [nm].
+
+    rdist: :class:`~xarray.DataArray`
+        Particle radius probability distribution.
+
+    m: :class:`~xarray.DataArray`
+        Complex refractive index.
+
+    mu: :class:`~pint.Quantity` or :class:`~numpy.ndarray`
+        Scattering angle cosines [dimensionless].
+
+    Returns
+    -------
+    :class:`~xarray.DataArray`
+        Radiative properties.
```

Comment:
We have built-in aliases for many of these (see `docs/conf.py`). In that case:
```suggestion
    w : quantity or array-like
        Wavelength values [nm].

    rdist : DataArray
        Particle radius probability distribution.

    m : DataArray
        Complex refractive index.

    mu : quantity or array-like
        Scattering angle cosines [dimensionless].

    Returns
    -------
    DataArray
        Radiative properties.
```

Reply:
Nice, thanks!